src/types/from_sql.rs

```diff
@@ -96,6 +96,15 @@ macro_rules! from_sql_integral(
                 i.try_into().map_err(|_| FromSqlError::OutOfRange(i))
             }
         }
+    );
+    (non_zero $nz:ty, $z:ty) => (
+        impl FromSql for $nz {
+            #[inline]
+            fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+                let i = <$z>::column_result(value)?;
```

Comment:
This seems wrong to me because SQLite supports only two kinds of integer (http://sqlite.org/c3ref/value_blob.html):
> sqlite3_value_int	→	32-bit INTEGER value
> sqlite3_value_int64	→	64-bit INTEGER value

And `rusqlite` actually uses only one: `sqlite3_value_int64`
So `<$z>` should always be `i64`.